XenAdmin/Dialogs/VmSnapshotDialog.cs

```diff
@@ -178,10 +178,10 @@ private void CheckpointInfoPictureBox_Click(object sender, EventArgs e)
                 tt = Messages.FIELD_DISABLED;
             else if (_VM.power_state != vm_power_state.Running)
                 tt = Messages.INFO_DISKMEMORY_MODE_POWER_STATE.Replace("\\n", "\n");
-            else if (_VM.HasVGPUs())
-                tt = Messages.INFO_DISKMEMORY_MODE_GPU.Replace("\\n", "\n");
             else if (!_VM.GetVirtualisationStatus().HasFlag(VM.VirtualisationStatus.IO_DRIVERS_INSTALLED))
                 tt = (_VM.HasNewVirtualisationStates() ? Messages.INFO_DISKMEMORY_MODE_NO_IO_DRIVERS : Messages.INFO_DISKMEMORY_MODE_NO_TOOLS).Replace("\\n", "\n");
+            else if (_VM.HasVGPUs())

```

Comment:
I don't know what other conditions might restrict the checkpoint operations, but the fact that we have the INFO_DISKMEMORY_MODE_MISC message suggests that it might be other reasons why the checkpoints are restricted. 
I suggest to either get rid of the INFO_DISKMEMORY_MODE_GPU message completely (so the message will be INFO_DISKMEMORY_MODE_MISC  even if the vGPU is the actual reason), or enhance the GPU check to also include a version check and the vGPU type (i.e. show the  INFO_DISKMEMORY_MODE_GPU if _VM.HasVGPUs and the server version is pre-Naples or the gpu type is not nvidia)

Reply:
Thanks.
Now I added new function to check the VGPU type is `Nvidia` or not and check the server version.
Please help to check again. @MihaelaStoica 
Also, after vgpu_type check and version check, I think the check order should still be checking the I/O drivers firstly rather than check VGPU.

Reply:
I don't think it is correct that the HasVGPUs() function returns false if the host is NaplesOrGreater and vGPU type is Nvidia ; it doesn't return what the name suggests.

Anyway, I think that the check on the server side (for allowing suspend and checkpoint operations) is a bit more complex, as only *some* Nvidia GPUs support suspending and checkponting a VM (by providing compatibility metadata).

Given this complexity, I don't think it's worth having version and vGPU type checks here, and we can simply display the generic message (INFO_DISKMEMORY_MODE_MISC). Please note that we have done the same for vGPU migration (see CA-270505).
And I believe we also have to remove the message for VM suspend (https://github.com/xenserver/xenadmin/blob/master/XenAdmin/Commands/SuspendVMCommand.cs#L222), for the same reason - some vGPU types allow VM suspend.

Reply:
Understood. 
I've removed the specific error message to generic message for VM checkpoint and suspend with two separate commits. 
Please take a look, thank you! :) @MihaelaStoica